aptos-move/move-examples/wrapped_token/sources/wrapped_token.move

```
@@ -0,0 +1,110 @@
+        burn_ref: Option<0x4::token::BurnRef>,
+        /// Used to control freeze.
+        transfer_ref: Option<object::TransferRef>,
+        /// Used to mutate fields
+        mutator_ref: Option<0x4::token::MutatorRef>,
+        /// Used to mutate fields
+        extend_ref: Option<ExtendRef>,
+    }
+
+    /// Create a collection based on the previous collection
+    ///
+    /// TODO: what should we do with the original collection?
+    public entry fun create_wrapped_collection(
+        new_creator: &signer,
+        creator_address: address,
+        name: String,
+    ) {
+        // TODO: Relax later?
+        assert!(signer::address_of(new_creator) == creator_address, 12345);
+        let description = aptos_token::token::get_collection_description(creator_address, name);
+        let uri = aptos_token::token::get_collection_uri(creator_address, name);
+        let max_supply = aptos_token::token::get_collection_maximum(creator_address, name);
+
+        // For now, allow mutable everything, TODO: Make this configurable / use previous state
+        0x4::aptos_token::create_collection(
+            new_creator,
+            description,
+            max_supply,
+            name,
+            uri,
+            true,
+            true,
+            true,
+            true,
+            true,
+            true,
+            true,
+            true,
+            true,
+            0,
+            1
+        );
+    }
+
+    /// This allows you to wrap based on the creator, though for consistency, you'd want the same collection and creator
+    public fun wrap(
+        creator: &signer,
+        receiver: address,
+        collection: String,
+        name: String,
+    ) {
+        let creator_address = signer::address_of(creator);
+        let data_id = aptos_token::token::create_token_data_id(creator_address, collection, name);
+        let token_id = aptos_token::token::create_token_id(data_id, 0);
+        // Load token TODO: handle fungible?
```

Let's just throw an error here if the v1 token is fungible to make it clear this example does not support fungible tokens yet